third-party/qthread/qthread-1.10/src/interfaces/chapel/tasks-qthreads.c

```diff
@@ -162,77 +162,39 @@ void chpl_task_yield(void)
 void chpl_sync_lock(chpl_sync_aux_t *s)
 {
     aligned_t l;
+    chpl_bool uncontested_lock = true;
 
     PROFILE_INCR(profile_sync_lock, 1);
 
-    l = qthread_incr(&s->lockers_in, 1);
-
-    // Yield if somebody else has the lock. If the current task can't get the
-    // lock then it can't make progress until the lock is released, so let
-    // somebody else do some work. It might be better to yield only after X
-    // number of failures to acquire the lock for cases where the lock would be
-    // released "soon", but we didn't see any performance impacts of doing it
-    // every time.
     //
-    // A core issue is that it's possible that a task scheduled on the same
-    // worker could be the one responsible for unlocking the sync var. For
-    // instance if a task locked the sync var, then does a chpl_task_yield, it
-    // could now be scheduled to run after the current task so this task must
-    // yield in order to prevent deadlock. However, if the task that will
-    // unlock is on another worker, then a spinloop would result in lower
-    // latency. That tension between deadlock avoidance and latency
-    // minimization is what pushes us to even consider spinning in addition to
-    // yielding, and to worry about how much of it we should do.
+    // To prevent starvation due to never switching away from a task
+    // that is spinning while doing readXX() on a sync variable, yield
+    // if this sync var has a "lot" of uncontested locks. Note that the
+    // uncontested locks do not have to be consecutive. Also note that
+    // the number of uncontested locks is a lossy counter. Currently a
+    // "lot" is defined as ~100 uncontested locks, with care taken to
+    // note yield on the first uncontested lock.
     //
-    // If real qthreads sync vars were used, it's possible this wouldn't be
-    // needed.
```

Comment:
You might consider leaving this in, perhaps rephrased. Something like:
If Chapel sync vars were native Qthreads sync vars, rather than derivations thereof, it's possible this would not be necessary. It depends on whether and/or how the internal Qthreads code does starvation avoidance.